### ISSUE TITLE
python 3.5 support for build system

### ISSUE THREAD

**Issue by pohmelie** [rtfd/readthedocs.org#1990]
Since build system use cpython 3.4 I can't build documentation for my 3.5 only project. https://readthedocs.org/projects/aioftp/builds/3722703
Is there some plans about 3.5?

**Comment by jakirkham** [comment]
+1

**Comment by agjohnson** [comment]
Python 3.5 will definitely come with the april ubuntu 16.04 release, we haven't discussed an interim plan -- though installing from the deadsnakes ppa could be an easy path forward. Another option is to use conda

**Comment by ian-ross** [comment]
+1 from me too: https://readthedocs.org/projects/django-tutelary/builds/3795496/ (problem caused by `yield from` in my case).

**Comment by agjohnson** [comment]
Conda support can target python 3.5, that might be the best bet.

**Comment by jakirkham** [comment]
I expect there will be the same situation when Python 3.6 comes around too. So, using `conda` seems reasonable for this problem.

**Comment by AndreLouisCaron** [comment]
On many Linux distros and on OSX, multiple versions of Python are often aliased to `python3.4`, `python3.5`, etc.

Can't we add an option in to select one of these?  We can keep the "CPython 2.x" and "CPython 3.x" options, but add "CPython 2.7", "CPython 3.4" and "CPython 3.5" in the choices too.

**Comment by agronholm** [comment]
+1, whatever solution works :) I have plenty of 3.5 only projects coming that I'd like to build docs for.

**Comment by evertheylen** [comment]
If people are too lazy to set up a conda environment; just use this configuration file: http://stackoverflow.com/a/36144269/2678118

**Comment by agronholm** [comment]
Question: how am I supposed to get the project installed in the conda environment?

**Comment by agjohnson** [comment]
We could use some help beta testing ubuntu 16.04 builders, which will include python 3.5. If anyone is interested, we're collecting a handful of testers here:

https://docs.google.com/forms/d/1YHBXN8LI2RTq6ujV0zENAExtJwbNLja0QKJUHyfUvrE/viewform

**Comment by algorys** [comment]
+1 for my project too. For https://readthedocs.org/projects/pycmake/builds/4247460/

**Comment by AndreLouisCaron** [comment]
FWIW, I now host my docs on [Python Hosted](http://pythonhosted.org) solely because of this issue, but the flow there is manual, error prone and doesn't easily allow me to host docs for multiple versions, which is really annoying.  I'd really appreciate if this were fixed...

Let me know if there's anything I can do to help.

**Comment by jakirkham** [comment]
Does the [answer]( https://github.com/rtfd/readthedocs.org/issues/1990#issuecomment-214739804 ) given by @evertheylen not work for you @AndreLouisCaron?

**Comment by AndreLouisCaron** [comment]
Sorry, must has missed that email!  Will try it ASAP :-)

André

**Comment by numberoverzero** [comment]
Any update?  I filled out the beta testing form above but haven't heard back.

**Comment by c-bata** [comment]
+1 my projects `typing` module.

**Comment by benjaminweb** [comment]
My docs are built without autodoc statements due to lack of Python 3.5:

`ImportError: cannot import name 'gcd'`

https://docs.python.org/3/library/math.html#math.gcd

What's the timeline?

**Comment by pohmelie** [comment]
I think it's time to support 3.6, since f-strings will be "syntax error" even for 3.5.

**Comment by ashwin31** [comment]
:+1:

**Comment by jagadeeshv16** [comment]
+1

**Comment by druuu** [comment]
+1

**Comment by chaitu210** [comment]
:+1:

**Comment by vidyasagar-r** [comment]
+1

**Comment by AnjaneyuluBatta505** [comment]
:+1:

**Comment by gvudayateja** [comment]
:+1:

**Comment by agronholm** [comment]
I'm helping the RTD staff create a better build image that allows for selecting the preferred Python version for the build. Python 3.5 and 3.6 are already included.

**Comment by vineeshan** [comment]
:+1:

**Comment by nikhila05** [comment]
:+1:

**Comment by swethanaretla** [comment]
:

**Comment by agronholm** [comment]
You can stop the reactions now, it's distracting and does not really help.

**Comment by agjohnson** [comment]
Just a quick update here, as I'm pushing out some updates this week.

First, through a pile of work, support for non-system Python versions has been added. The new beta build container image is now ready for testing (rtfd/readthedocs-docker-images#18). This will support versions: 2.7, 3.3, 3.4, 3.5, and 3.6. This image uses `pyenv` to install multiple versions of Python, instead of relying on the distribution's versions.

This will be the new beta image going forward. If you'd like to continue testing our beta images, or would like support for python 3.6, be sure to sign up here:
https://goo.gl/forms/ai9u6jpIhpEpmMRx1

If you've been looking for python 3.5 support, and have not been added to the beta list, we are going to be switching over everyone this week to an image with python 3.5 support. You shouldn't need to do anything, but adding a `readthedocs.yml` file specifying the correct python version might help if you run into problems.

I am going to write up some docs on using the feature tonight, we'll be announcing changes later this week.

**Comment by c-bata** [comment]
Great updates! I've just signed up the beta support for python3.6. 👍

**Comment by humitos** [comment]
This issue was create to support Python 3.5 and it was added in the `readthedocs/build:2.0` image so I think the work on this issue is already done and working properly (I saw a many projects using python3.5 without problem).

Shouldn't we close this one and move the discussion regarding Python 3.6 to another one?

**Comment by Granitosaurus** [comment]
For anyone who ends up here for py3.6 support it's officially available:
http://blog.readthedocs.com/python-36-support/

TL;DR:
create file `readthedocs.yaml` with contents in your project root:

    build:
        image: latest

    python:
        version: 3.6

**Comment by RichardLitt** [comment]
I wonder if there is a place in the documentation to include this note from @Granitosaurus. @humitos What do you think?
